staging/src/k8s.io/client-go/tools/leaderelection/leaderelection.go

```diff
@@ -318,14 +318,27 @@ func (le *LeaderElector) release() bool {
 // on success else returns false.
 func (le *LeaderElector) tryAcquireOrRenew(ctx context.Context) bool {
 	now := metav1.NewTime(le.clock.Now())
+	// 1. fast path to update optimistically assuming that the record observed
+	// last time is the current version.
+	oldObservedRecord := le.getObservedRecord()
+	if le.IsLeader() && le.observedRecordOnLease(now.Time) {
+		oldObservedRecord.RenewTime = now
```

Comment:
Let's not modify `oldObservedRecord` - this is super misleading.

You should leave creation of `leaderElectionRecord` from below as it was and just use that and update the fields as in lines 373-374.